presto-main/src/main/java/com/facebook/presto/operator/TableFinishInfo.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.connector.ConnectorOutputMetadata;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.json.JsonCodec;
+
+import java.util.Optional;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+
+public class TableFinishInfo
+        implements Mergeable<TableFinishInfo>
+{
+    private static final JsonCodec<ConnectorOutputMetadata> CONNECTOR_OUTPUT_METADATA_CODEC = jsonCodec(ConnectorOutputMetadata.class);
+
+    private Optional<String> connectorOutputMetadata;
+
+    static TableFinishInfo tableFinishInfo(Optional<ConnectorOutputMetadata> connectorOutputMetadata)
+    {
+        return new TableFinishInfo(connectorOutputMetadata.map(CONNECTOR_OUTPUT_METADATA_CODEC::toJson));
+    }
+
+    @JsonCreator
+    public TableFinishInfo(
+            @JsonProperty("connectorOutputMetadata") Optional<String> connectorOutputMetadata)
+    {
+        this.connectorOutputMetadata = connectorOutputMetadata;
+    }
+
+    @JsonProperty
+    public Optional<String> getConnectorOutputMetadata()
+    {
+        return connectorOutputMetadata;
+    }
+
+    @Override
+    public TableFinishInfo mergeWith(TableFinishInfo other)
+    {
+        if (other.getConnectorOutputMetadata().isPresent()) {
```

Comment:
Under what scenario would the engine try to merge two TableFinishInfo? There should be a single one per query, no?